spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -240,8 +245,38 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
     }
   }
 
+  public boolean isHiveCatalogConfigValid(CaseInsensitiveStringMap options, StringBuilder errorMsg) {
+    String hadoopConfUri = SparkSession.active().sparkContext().conf().contains("spark.hadoop.hive.metastore.uris") ?
+            SparkSession.active().sparkContext().conf().get("spark.hadoop.hive.metastore.uris") : null;
+    String catalogConfUri = options.get("uri");
+
+    if (hadoopConfUri == null) {
```

Comment:
After checking the `catalogHiveUri` is non-null, I think the only check that is necessary is whether it is equal to the URI from the environment:

```java
Preconditions.checkArgument(catalogHiveUri.equals(envHiveUri),
    "Invalid option, uri is not supported for SparkSessionCatalog: %s", catalogHiveUri);
```